lib/client.js

```diff
@@ -228,6 +228,13 @@ extend(Raven.prototype, {
       values: domainContext.breadcrumbs || this._globalContext.breadcrumbs || []
     };
 
+    kwargs.request = extend({}, this._globalContext.request, domainContext.request, kwargs.request);
+    if (Object.keys(kwargs.request).length === 0) {
+      var req = extend({}, this._globalContext.req, domainContext.req, kwargs.req);
+      var parseUser = Object.keys(kwargs.user).length === 0 ? this.parseUser : false;
+      kwargs = extend(kwargs, parsers.parseRequest(req, parseUser));
```

Comment:
extend mutates kwargs without the assignment but if you find this clearer its up to you